bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/Clip2ThingHandler.java

```diff
@@ -0,0 +1,1153 @@
+     */
+    public void onResourcesList(ResourceType resourceType, List<Resource> fullResources) {
+        if (resourceType == ResourceType.SCENE) {
+            updateSceneContributors(fullResources);
+        } else {
+            fullResources.stream().filter(r -> resourceId.equals(r.getId())).findAny()
+                    .ifPresentOrElse(r -> onResource(r), () -> {
+                        if (resourceType == thisResource.getType()) {
+                            logger.debug("{} -> onResourcesList() configuration error: unknown resourceId", resourceId);
+                            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                                    "@text/offline.api2.conf-error.resource-id-bad");
+                        }
+                    });
+        }
+    }
+
+    /**
+     * Process the incoming Resource to initialize the alert channel.
+     *
+     * @param resource a Resource possibly with an Alerts element.
+     */
+    private void updateAlertChannel(Resource resource) {
+        Alerts alerts = resource.getAlerts();
+        if (Objects.nonNull(alerts)) {
+            List<StateOption> stateOptions = alerts.getActionValues().stream().map(action -> action.name())
+                    .map(actionId -> new StateOption(actionId, actionId)).collect(Collectors.toList());
+            if (!stateOptions.isEmpty()) {
+                stateDescriptionProvider.setStateOptions(new ChannelUID(thing.getUID(), CHANNEL_2_ALERT), stateOptions);
+                logger.debug("{} -> updateAlerts() found {} associated alerts", resourceId, stateOptions.size());
+            }
+        }
+    }
+
+    /**
+     * If this v2 thing has a matching v1 legacy thing in the system, then for each channel in the v1 thing that
+     * corresponds to an equivalent channel in this v2 thing, and for all items that are linked to the v1 channel,
+     * create a new channel/item link between that item and the respective v2 channel in this thing.
+     */
+    private void updateChannelItemLinksFromLegacy() {
+        if (!disposing) {
+            legacyLinkedChannelUIDs.forEach(legacyLinkedChannelUID -> {
+                String targetChannelId = REPLICATE_CHANNEL_ID_MAP.get(legacyLinkedChannelUID.getId());
+                if (Objects.nonNull(targetChannelId)) {
+                    Channel targetChannel = thing.getChannel(targetChannelId);
+                    if (Objects.nonNull(targetChannel)) {
+                        ChannelUID uid = targetChannel.getUID();
+                        itemChannelLinkRegistry.getLinkedItems(legacyLinkedChannelUID).forEach(linkedItem -> {
+                            String item = linkedItem.getName();
+                            if (!itemChannelLinkRegistry.isLinked(item, uid)) {
+                                logger.info(
+                                        "{} -> updateChannelItemLinksFromLegacy() created link between Channel:{} and Item:{}",
```

Comment:
This should also be made more user-oriented since even **info** level.

Reply:
Ok. Done.